presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeIdenticalWindows.java

```diff
@@ -0,0 +1,108 @@
+import com.google.common.collect.ImmutableMap;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import javax.inject.Provider;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.any;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.window;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestMergeIdenticalWindows
+{
+    private final LocalQueryRunner queryRunner;
+
+    public TestMergeIdenticalWindows()
+    {
+        this.queryRunner = new LocalQueryRunner(testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .build());
+
+        queryRunner.createCatalog(queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(queryRunner.getNodeManager(), 1),
+                ImmutableMap.<String, String>of());
+    }
+
+    @Test
+    public void testFoo()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "sum(quantity) over (partition by orderkey order by shipdate rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertPlan(sql,
+                any(
+                        any(
+                                window(ImmutableList.of(fc("sum", "quantity"), fc("sum", "discount")),
+                                        window(ImmutableList.of(fc("sum", "quantity")),
+                                                any())))));
+    }
+
+    private FunctionCall fc(String name, String... args) {
+        ImmutableList.Builder<Expression> builder = ImmutableList.builder();
+        for (String arg : args) {
+            builder.add(new QualifiedNameReference(new QualifiedName(arg)));
+        }
+
+        return new FunctionCall(new QualifiedName(name), builder.build());
+    }
+
+    private void assertPlan(@Language("SQL") String sql, PlanMatchPattern pattern)
+    {
+        Plan actualPlan = plan(sql);
+        queryRunner.inTransaction(transactionSession -> {
+            PlanAssert.assertPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    private Plan plan(@Language("SQL") String sql)
+    {
+        FeaturesConfig featuresConfig = new FeaturesConfig()
+                .setExperimentalSyntaxEnabled(true)
+                .setDistributedIndexJoinsEnabled(false)
```

Comment:
Is this meaningful for this rewriter? If not maybe skip it or at least put comment next to it?